### PR TITLE
[JENKINS-36502] fixed compatibility with Pipeline Snippet Generator

### PR DESCRIPTION
**Pull request by daspilker**

### CODE REVIEW

**Comment by bitwiseman** [comment]
Instead of adding symbol to the deprecated interface, why not add it to this new path?

**Comment by bitwiseman** [comment]
This is great! 

Remember to update the power moves for the `jobDsl` pipeline step. 
https://github.com/jenkinsci/job-dsl-plugin/commit/491249b6e743daff322dbcc56e0ad9b75d5f1d09#diff-679b049f871204f96aec7b6536fdd272R217

Also, I'd suggest documenting the `jobDsl` pipeline step as the default way to call this, and noting the `step` syntax for those using Jenkins pre-2.10.

File: job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java [hunk 1]
```diff
@@ -191,18 +194,38 @@ public String getTargets() {
         return targets;
     }
 
+    @DataBoundSetter
+    public void setTargets(String targets) {
+        this.targets = StringUtils.isBlank(targets) ? null : targets;
```
Comment (bitwiseman) [inline job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java]: Consider using [hudson.Util.fixEmpty(string)](http://javadoc.jenkins-ci.org/hudson/Util.html#fixEmpty-java.lang.String-) for the exact same behavior, or [hudson.Util.fixEmptyAndTrim(string)](http://javadoc.jenkins-ci.org/hudson/Util.html#fixEmptyAndTrim-java.lang.String-) for cleaning up whitespace only input.

File: job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java [hunk 2]
```diff
@@ -191,18 +194,38 @@ public String getTargets() {
...
+    }
+
     public String getScriptText() {
         return scriptText;
     }
 
+    @DataBoundSetter
+    public void setScriptText(String scriptText) {
+        this.scriptText = StringUtils.isBlank(scriptText) ? null : scriptText;
```
Comment (bitwiseman) [inline job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java]: Consider using [hudson.Util.fixEmpty(string)](http://javadoc.jenkins-ci.org/hudson/Util.html#fixEmpty-java.lang.String-) for the exact same behavior.

File: job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java [hunk 3]
```diff
@@ -245,14 +268,28 @@ public String getAdditionalClasspath() {
 
     @DataBoundSetter
     public void setAdditionalClasspath(String additionalClasspath) {
-        this.additionalClasspath = additionalClasspath;
+        this.additionalClasspath = StringUtils.isBlank(additionalClasspath) ? null : additionalClasspath;
```
Comment (bitwiseman) [inline job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java]: Consider using [hudson.Util.fixEmpty(string)](http://javadoc.jenkins-ci.org/hudson/Util.html#fixEmpty-java.lang.String-) for the exact same behavior.

**Comment by daspilker** [comment]
Thanks for the review!

The snippet generator output includes the `usingScriptText` field of `ExecuteDslScripts`. But that's unnecessary, it only exists for the `radioblock` in `config.jelly`. Do you have an idea for avoiding that output?

**Comment by bitwiseman** [comment]
@daspilker  - Yes.   Similar to #880, [ExecuteDslScripts.java lines 97-113](https://github.com/jenkinsci/job-dsl-plugin/pull/880/files#diff-f43761d263ee3c46390f32891bca87f4R97)

In your config.jelly, instead of changing the name to "usingScriptText", change it to "scriptStyle". 
Then the following should work.   It may be counter to design guidelines, but it should work. 😄 


```
        // We want to be able to set this, but we never want it to return a value.
        // This will prevent the snippet generator generating output for this field,
        // while also not throwing an exception due to missing getter.
        public String getScriptStyle() {
            return null;
        }

        // This property is optional and one-directional (set only).
        // It is set only from the UI and will force usingScriptText to true or
        // false, based on the UI databound value.
        @DataBoundSetter
        public void setScriptStyle(String value) {
            value = Util.fixEmptyAndTrim(value);
            if (value != null) {
                this.usingScriptText = Boolean.parseBoolean(value);
            }
        }

        // If not forced, isUsingScriptText() will be calculated based on
        // the presence of non-null scriptText.
        public boolean isUsingScriptText() {
            if (usingScriptText == null) {
                return scriptText != null;
            } else {
                return usingScriptText;
            }
        }
```

Basically, `isUsingScriptText is the read-only side, and `scriptStyle` is the set only side that satisfies databinding.

File: docs/User-Power-Moves.md [hunk 4]
```diff
@@ -194,12 +209,12 @@ node {
...
-* `scriptLocation`: mandatory, specifies the Job DSL scripts to execute and consists of three sub-options:
-     * `targets`: optional, specifies Job DSL script files to execute, newline separated list of file names relative
-                  to the workspace
-     * `scriptText`: optional, specifies an inline Job DSL script
-     * `ignoreMissingFiles`: optional, defaults to `false`, set to `true` to ignore missing files or empty wildcards in
-                            `targets`
+* `targets`: optional, specifies Job DSL script files to execute, newline separated list of file names relative
+             to the workspace
+* `scriptText`: optional, specifies an inline Job DSL script
+* `usingScriptText`: optional, `scriptText` will be used if set to `true`, `targets` will be used if set to `false`
```
Comment (bitwiseman) [inline docs/User-Power-Moves.md]: Should be noted as defaulting to true.

**Comment by bitwiseman** [comment]
Here's a commit with some additional tweaks to make for better behavior.

**Comment by daspilker** [comment]
@bitwiseman Thanks, I picked your commit. I think this is ready to merge. Any objections?

**Pull request merged**